File: messaging-frontend/src/types/message.ts
import type { Agent } from "./agent"
import type { Conversation } from "./conversation"
import type { Customer } from "./customer"

export type Message = {
    id: string,
    content: string,
    timestamp: Date,
    replyTo: Message,
    fileUrl: string,
    customer: Customer,
    agent: Agent,
    edited: boolean,
    read: boolean,
    senderType: 'CUSTOMER' | 'AGENT',
    conversation: Conversation
}

export type MessageCreationRequest = {
    content: string,
    agentId: string,
    senderType: 'CUSTOMER' | 'AGENT',
    conversationId: string,
    replyToId?: string
}

File: MessagingAndroid/app/src/main/java/ma/tayeb/messaging_android/types/Message.kt
package ma.tayeb.messaging_android.types

import ma.tayeb.messaging_android.enums.SenderType
import org.threeten.bp.LocalDateTime
import java.util.UUID

data class Message(
    val id: String?,
    var content: String?,
//    val timestamp: LocalDateTime,
    val timestamp: String,
    val replyTo: Message?,
    val fileUrl: String?,
    val customer: Customer?,
    val agent: Agent?,
    val edited: Boolean,
    val read: Boolean,
    val senderType: SenderType,
    val conversation: Conversation
)

data class MessageCreationRequest(
    val content: String?,
    val customerId: UUID,
    val senderType: SenderType,
    val conversationId: UUID,
    val replyToId: String?
)